src/components/AssetIcon.tsx

```diff
@@ -63,12 +63,24 @@ export const AssetIcon = ({ assetId, showNetworkIcon, src, ...rest }: AssetIconP
 
   if (!asset && !assetId && !src) return null
 
+  if (assetId === nativeAssetId && asset?.networkIcon && showNetworkIcon) {
+    return (
+      <Avatar
+        src={asset.networkIcon}
+        bg={assetIconBg}
+        icon={<FoxIcon boxSize='16px' color={assetIconColor} />}
+        {...rest}
+      />
+    )
+  }
+
   if (assetId) {
-    if (asset && asset.icons) {
+    if (asset?.icons) {
```

Comment:
We'll want to be very careful with the flow change here and ensure pair / network icons show no regressions across the app, this is quite fragile currently

Reply:
Tested across `/assets`, asset, account asset and trade asset selection, didn't find any regressions against prod ✅